HybridCompute stable release version 2022-12-27

yaotongms opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

annelo-msft committed:
Reshuffle shared source files (#38503)

* Reshuffle shared source files

* Add

* missed files

* mutable json document

jalauzon-msft committed:
[Storage][DataMovement] Add classes for Job level checkpoint files, JobPlanFile (#38439)

annelo-msft committed:
Adding string cast operators and null APIs to Variant (#38508)

* Adding string cast operators and null APIs

* pr fb

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230905.4 (#38507)

live1206 committed:
Prepare release for AppService (#38512)

ankur741999 committed:
Updated autorest.md with swagger sdk (#38250)

* Updated autorest.md with swagger sdk

* Updated with latest set of commands

* Updated path

* Add flatted property back to mitigate breaking change

* Export API

* Added Renaming for securitysettings,softdeletesettings and softdeletestate, updated changelog

* Added chanelog

* Updating the mapping in custom code

* Export API

* Prepare release

---------

Co-authored-by: Wei Hu <huwe@microsoft.com>

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.AppService (#38513)

williamzhao87 committed:
Fix internal classes (#38520)

williamzhao87 committed:
Update changelog for beta.2 (#38523)

azure-sdk committed:
Increment package version after release of Azure.Communication.JobRouter (#38529)

jsquire committed:
[AppConfig] Preserve unknown setting JSON members (#38450)

* [AppConfig] Preserve unknown setting JSON members

The focus of these changes is to fix a bug where JSON members in the
value for Feature Flags and Secret References which are not defined in
the JSON schema were discarded.  The schema itself allows for arbitrary
JSON members, the Azure portal supports them in its advanced view, and
the service documentation also refers to them.

This change preserves unknown JSON attributes while allowing the well-known
properties of the setting to override their associated JSON members.

Constructor overloads that accept an ETag were also added to Feature Flags
and Secret References.  This aligns with ConfigurationSetting and allows use
of the GetConfigurationSettingsAsync overload that accepts `onlyIfUnchanged`,
which was previously not accessible for specialized settings.

christothes committed:
Fix docs for CreateTableIfNotExists return value (#38535)

annelo-msft committed:
Reshuffle shared source files (#38509)

* Move GZipUtf8JsonRequestContent to Monitor

* Move GuidUtilities to ServiceBus

* Make GeoRedundantFallbackPolicy internal

* move ErrorResponse<T> into HttpPipelineExtensions

* Move NoValueResponse<T> to Tables

Commit:
update changelog for crp 2023-07-01 (#38362)

emmeliaAra committed:
[HardwareSecurityModules] Generating SDK (#37490)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230906.1 (#38528)

rangothedog committed:
[Text Translation] Replace async examples with synchronous code. (#38386)

* Update Readme.md snippets to be populated by tool, added SampleSnippets.cs file as source.

* Fixing missing double quote on region identifier.

* Updating sample C# snippets and code (#2)

* Updating sample C# snippets and code.

* Use explicit type instead of var, remove blank lines from #if

* Finished change to use explicit type instead of var

---------

Co-authored-by: Rango Meadows <v-rameadows@microsoft.com>

* Replace async samples with synchronous code

* Corrected reference to async samples in README.md

---------

Co-authored-by: Rango Meadows <v-rameadows@microsoft.com>
Co-authored-by: Michal Materna <michal.materna@hotmail.com>

ankur741999 committed:
.NetSDK Release RSBackup 2023-04-01 (#38462)

* .NetSDK Release RSBackup 2023-04-01

* .NetSDK Release RSBackup 2023-04-01

* Updated changelog and added custom file for breaking change

* Ran provided commands to fix CI issues

* Revert "Ran provided commands to fix CI issues"

This reverts commit 9b470c969cdb00c42b561069be3a11df9dea5393.

* .NetSDK Release RSBackup 2023-04-01

* Updated changelog and added custom file for breaking change

* Ran provided commands to fix CI issues

* Revert "Ran provided commands to fix CI issues"

This reverts commit 9b470c969cdb00c42b561069be3a11df9dea5393.

* Ran commands to fix CI

* Added Custom Files for breaking change

* Added never attribute

* Updated changlog and csproj with correct version

* ran the export api command

* Updated changelog

* Resolved comment

dvbb committed:
fix uri format issue (#38552)

RodgeFu committed:
Rename configuration rename-rules to acronym-mapping (#38294)

Co-authored-by: Arthur Ma <arthurma@microsoft.com>

jsquire committed:
[AppConfig] Fix disposed stream access (#38531)

The focus of these changes is to fix a bug where the content stream returned
by the protocol method has been disposed and the serializer attempts to access
it.  Shifting to accessing the `BinaryData` content directly for deserialization.

christothes committed:
Identity Sept release prep (#38544)

arlynch committed:
Update to new api version 2023-06-30 (#36997)

* added changes for 2023-06-30 api release

* updated generated files

* fixed formatting in changelog

* Updated method names and property types per previous review comments

* Updated changelog to remove references to beta

* Set InnerError to be internal Class

* Fixed error related to changelog versions

* Removed the ability to publicly set error in the import job model

* Updated session records

* Updated Apis

---------

Co-authored-by: Austin Lynch <austinlynch@microsoft.com>

annelo-msft committed:
Update changelog for release (#38563)

annelo-msft committed:
Update changelog and version for release (#38562)

ShivangiReja committed:
[App Config] Rename snapshot properties (#38540)

annelo-msft committed:
Implement ToString for Variant (#38561)

* Implement ToString

* use if/else

* Export API

* Update changelog for release

* update changelog

* bools

* pr fb

* nit

* nit

williamzhao87 committed:
[JobRouter] Add getter for ScoringParameters (#38564)

Commit:
[AzureMonitorExporter] update OTel to 1.6.0 (#38568)

* update OTel to 1.6.0

* changelog

jaschrep-msft committed:
DataMovement StorageResource construction refactor | Part 4 (#38534)

* hid implementation classes

* exportapi | updatesnippets

azure-sdk committed:
Increment version for core releases (#38570)

* Increment package version after release of Azure.Core

* Increment package version after release of Azure.Core.Experimental

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 6903 (#38510)

* fixes

* ongoing

* ongoing

* add top-level throw/catches showing exception info

* fix handling of cases when there is 1 link and when there is no RetryAfter.Delta

* handle lack of Exception.Headers property

* handle gracefully obtaining status code from $_.Exception.InnerException.ErrorCode

---------

Co-authored-by: Konrad Jamrozik <kojamroz@microsoft.com>

annelo-msft committed:
Update Azure.Core version and package refs (#38572)

arlynch committed:
updated changelog (#38578)

Co-authored-by: Austin Lynch <austinlynch@microsoft.com>

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.RecoveryServicesBackup (#38584)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230907.1 (#38558)

slpavkov committed:
3398502 - Fix ACS PSTN Sip Configuration SDK tests (#38588)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230907.2 (#38559)

jsquire committed:
[AppConfig] Reset development version (#38577)

With the needed release changes pushed to the hotfix branch, this resets
the development version and change log to guard against accidental release.

arlynch committed:
fixed method name in sampple helper (#38594)

Co-authored-by: Austin Lynch <austinlynch@microsoft.com>

alrod committed:
Make scale methods per trigger internal (#38581)

christothes committed:
Fix for first run cache miss for interactive credentials (#38449)

RodgeFu committed:
Include sample-gen by default when generating sdk code (mgmt) (#38277)

* update autorest.md to include sample-gen config and delete autorest.tests.md

* update more config

* update config of hardwaresecuritymodule

* fix blueprint sample config

* update to include skip sample config

* temp remove samplegen config for services using batch config

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.RecoveryServices (#38583)

Y-Sindo committed:
Release v1.11.1 (#38602)

jsquire committed:
[Event Hubs] September 2023 Release Prep (#38604)

The focus of these changes is to prepare the core Event Hubs library for
the September 2023 release.

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools repository (#38606)

annelo-msft committed:
Don't allow setting Variant to Variant (#38592)

* Don't allow setting Variant to Variant

* pr fb

annelo-msft committed:
Allow As<T> to return null for reference types (#38611)

* Allow As<T> to return null for reference types

* Improve cast exception message

Joeleniqs committed:
[Communication]Enable managed identity in ACScommunication resource sdk (#37566)

* Update autorest mark down to latest OpenApi spec

* initial code generation with new Swagger spec hash

* remove bugged format and generate code

* update spec link to remote link

* Adding test scenario for MI

* Update Export Api

* Update to most recent spec hash

* Update Sample Snippets

* Add Managed Identity tests session records assets

* Re-enforce resourceId format

* updated test recording

* updated recordings

* updated-recordings

* fix tests and Update test recordings

* Update changelog release history

* Provide backward compatibility to accomodate breaking change

* remove param from fallback overload

* Update customize folder with communication service resource data

* fix build issues

* fix build issues

* export api to support older resource data

* fix build issues

* Update release date of `1.2.0-beta.2`

---------

Co-authored-by: Joel Enikanorogbon <joelen@microsoft.com>
Co-authored-by: Jorge Beauregard <jbeauregardb@microsoft.com>
Co-authored-by: Mingzhe Huang <archerzz@users.noreply.github.com>

azure-sdk committed:
Increment package version after release of Microsoft.Azure.WebJobs.Extensions.SignalRService (#38618)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Communication (#38617)

azure-sdk committed:
Increment package version after release of Azure.Messaging.EventHubs (#38627)

jsquire committed:
[Event Hubs] September 2023 Release Prep (non-core) (#38605)

The focus of these changes is to prepare the Event Hubs processor and
function extensions libraries for the September 2023 release.

annelo-msft committed:
Remove circular shared source references in Azure.Core tests files (#38607)

* Initial reshuffle

* Second reshuffle

* nits

* fix

Y-Sindo committed:
[SignalR extensions] Release v1.11.2 (#38623)

* Release v1.11.2

* Fix test issue

seanmcc-msft committed:
STG 90 GA release prep (#38631)

christothes committed:
Fix release date for Confidential Ledger (#38629)

azure-sdk committed:
Increment version for eventhub releases (#38632)

* Increment package version after release of Microsoft.Azure.WebJobs.Extensions.EventHubs

* Increment package version after release of Azure.Messaging.EventHubs.Processor

jsquire committed:
[AppConfig] Update API Compat version (#38628)

The focus of these changes is to update the API Compat target after the 1.2.1 release.

JoshLove-msft committed:
Prepare for release (#38612)

* Prepare for release

* change log

azure-sdk committed:
Increment package version after release of Azure.Communication.JobRouter (#38641)

azure-sdk committed:
Increment version for storage releases (#38642)

ArcturusZhang committed:
revert some unexpected sample configuration changes (#38619)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230912.3 (#38653)

azure-sdk committed:
Increment package version after release of Azure.Security.ConfidentialLedger (#38644)

christothes committed:
update version to patch (#38646)

christothes committed:
Update dockerfile for net6.0 (#38645)

ReubenBond committed:
NonDisposingStream: delegate remaining Stream APIs to inner stream (#38184)

ShivangiReja committed:
[App Config] Snapshot API Updates Based on Arch Board Feedback (#38650)

 [App Configuration] Updates in snapshot API

JoshLove-msft committed:
Fix casing on some events (#38674)

alrod committed:
Legacy checkpoint support (#38546)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 6895 (#38672)

* Add legacy moniker migration logic

* Add functionality from Java testing

---------

Co-authored-by: Daniel Jurek <djurek@microsoft.com>

danieljurek committed:
.NET-specific changes: Move deprecated packages to legacy (#36194)

* First cut at migrating to legacy moniker

* Update-PackageMonikers.ps1

* Refinement

* Update docindex.yml for testing

* Rename

* ToC fixes

* Refactor namespace discovery

* Remove EnsureNamespaces

* Add "legacy" moniker

* Undo Service-Level-Readme-Automation.ps1 because it is generating incorrect package paths

* Better edge case handling, documentation

* Add fixes to service level overview generation

* Reset eng/common

* docindex changes

* Comment cleanup

* Fix ToC generation tests

* Test cleanup

annelo-msft committed:
Remove unused <IncludeGeneratorSharedCode> MSBuild properties (#38648)

* Remove <IncludeGeneratorSharedCode> MSBuild properties

* revert personalizer

azure-sdk committed:
Increment package version after release of Azure.Messaging.EventGrid (#38682)

rangothedog committed:
[Document Translation] Fix Document Translation README.md links (#38593)

* Fix Document Translation README.md links

* Remove extra whitespace and remove en-us from links

* Revert mocking link change

* Remove duplicate URL

---------

Co-authored-by: Rango Meadows <v-rameadows@microsoft.com>

live1206 committed:
Prepare release for AppService (#38696)

JialinXin committed:
[WebPubSub] Increase isolated package version and fix a few issues (#38692)

* Increase isolated process package.

* Fix title

alrod committed:
Fixing scaler provider initialization pattern (#38673)

jaschrep-msft committed:
src, test, and samples projects for DataMovement.Files.Shares (#38678)

* src, test, and samples projects for DataMovement.Files.Shares

* placeholder readme

* remove shared source

* more csproj

* blank changelog

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.ResourceManager.HybridCompute](https://apiview.dev/Assemblies/Review/c04b4b8d01ef44d4aa49ef42dc0ff209)

vishweshbankwar committed:
Remove metric telemetry for resource (#38697)

* remove metric telemetry for resource

* distro test

* rename env var

* test

* update test

* const

vishweshbankwar committed:
Minor fix (#38724)

jalauzon-msft committed:
[Storage][DataMovement] Integrate JobPlanFile with checkpointer (#38595)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230913.2 (#38704)

Commit:
Fix resource provider namespace value (#38730)

live1206 committed:
Fix string format for serialization (#38699)

* Fix string format for StartOn

* update changelog

* Apply SerializationValueHook to simplify

* prepare release

archerzz commented on sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs:
```diff
@@ -32,21 +37,135 @@ public HybridComputeMachineData(AzureLocation location) : base(location)
+        /// <summary> The resource id of the parent cluster (Azure HCI) this machine is assigned to, if any. </summary>
+        public string ParentClusterResourceId { get; set; }
+        /// <summary> Specifies whether any MS SQL instance is discovered on the machine. </summary>
+        public string MssqlDiscovered { get; set; }
```
Should that be `MsSqlDiscovered`?
```suggestion
        public string MsSqlDiscovered { get; set; }
```

archerzz commented on sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs:
```diff
@@ -32,21 +37,135 @@ public HybridComputeMachineData(AzureLocation location) : base(location)
+        /// <summary> Specifies the DNS fully qualified display name. </summary>
+        public string DnsFqdn { get; }
+        /// <summary> The resource id of the private link scope this machine is assigned to, if any. </summary>
+        public string PrivateLinkScopeResourceId { get; set; }
```
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

archerzz replied:
Same question for `ParentClusterResourceId`.

yaotongms replied:
@archerzz I got an error opening the example links although being a member of the microsoft org. Wonder if you can send out another link or screenshot for the ResourceIdentifier example.

![image](https://github.com/Azure/azure-sdk-for-net/assets/137821665/2715dbc1-e1f5-420b-801a-caaecd7af199)

archerzz commented on sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs:
```diff
@@ -32,21 +37,135 @@ public HybridComputeMachineData(AzureLocation location) : base(location)
+        /// <summary> The type of Operating System (windows/linux). </summary>
+        public string OSType { get; set; }
+        /// <summary> Specifies the Arc Machine's unique SMBIOS ID. </summary>
+        public string VmUuid { get; }
```
Can we change the type to [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0)? See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).

yaotongms replied:
changed the type of VmUuid to Guid.

archerzz commented on sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/HybridComputeMachineData.cs:
```diff
@@ -32,21 +37,135 @@ public HybridComputeMachineData(AzureLocation location) : base(location)
+        /// <summary> The hybrid machine agent full version. </summary>
+        public string AgentVersion { get; }
+        /// <summary> Specifies the hybrid machine unique ID. </summary>
+        public string VmId { get; set; }
```
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

yaotongms replied:
changed the type of VmId to Guid.

archerzz commented on sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs:
```diff
@@ -0,0 +1,48 @@
+        /// <summary> Specifies the version info w.r.t AgentUpgrade for the machine. </summary>
+        public string DesiredVersion { get; set; }
+        /// <summary> The correlation ID passed in from RSM per upgrade. </summary>
+        public string CorrelationId { get; set; }
```
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).

yaotongms replied:
changed the type of CorrelationId to Guid.

archerzz commented on sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/AgentUpgrade.cs:
```diff
@@ -0,0 +1,48 @@
+        /// <summary> Specifies if RSM should try to upgrade this machine. </summary>
+        public bool? EnableAutomaticUpgrade { get; set; }
+        /// <summary> Timestamp of last upgrade attempt. </summary>
+        public string LastAttemptTimestamp { get; }
```
Could this be a type of `DateTimeOffset`? See an example [here](https://github.com/Azure/azure-sdk-for-net/blob/e4fce1607d65d709b19c81cc78969cd08b804244/sdk/hdinsight/Azure.ResourceManager.HDInsight/src/autorest.md?plain=1#L146).

yaotongms replied:
changed the type of LastAttemptTimestamp to DateTimeOffset.

archerzz commented on sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/HybridComputeMachinePatch.cs:
```diff
@@ -19,7 +19,23 @@ public HybridComputeMachinePatch()
+        /// <summary> The info of the machine w.r.t Agent Upgrade. </summary>
+        public AgentUpgrade AgentUpgrade { get; set; }
+        /// <summary> The resource id of the parent cluster (Azure HCI) this machine is assigned to, if any. </summary>
+        public string ParentClusterResourceId { get; set; }
```
Same question regarding the format of the ID

yaotongms replied:
changed the type of ParentClusterResourceId to Guid.

yaotongms replied:
fixed in the latest commit

archerzz commented on sdk/hybridcompute/Azure.ResourceManager.HybridCompute/src/Generated/Models/MachineAssessPatchesResult.cs:
```diff
@@ -0,0 +1,67 @@
+        /// <summary> The overall success or failure status of the operation. It remains "InProgress" until the operation completes. At that point it will become "Unknown", "Failed", "Succeeded", or "CompletedWithWarnings.". </summary>
+        public PatchOperationStatus? Status { get; }
+        /// <summary> The activity ID of the operation that produced this result. </summary>
+        public string AssessmentActivityId { get; }
```
same question for the format of ID

yaotongms replied:
changed the type of AssessmentActivityId to Guid.

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230914.1 (#38733)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Compute (#38742)

azaslonov committed:
Initial version of API Center .NET SDK. (#38249)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ApiCenter (#38745)

kinelski committed:
[FormRecognizer] Tests: implemented model caching (#38643)

azure-sdk committed:
Increment package version after release of Microsoft.Azure.WebJobs.Extensions.SignalRService (#38738)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Maintenance (#38734)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.AppService (#38703)

shadahmed-microsoft committed:
Azure Defender EASM Management Plane SDK Public Preview (#38158)

aditimalladi committed:
Generate Appliance ResourceConnector dotnet SDK (#38308)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.DefenderEasm (#38748)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ResourceConnector (#38747)

JoshLove-msft committed:
Enforce API Compat (#38717)

* Enforce running ApiCompat

* Remove runapicompat property from storage extensions

* Add metapackage exclusion

* Only validate that RunApiCompat is not false for specific target

* fix

* temporary test to make sure CI fails

* Use Build step instead of Analyze

* Use parameter

* PR fb

azure-sdk committed:
update proxy version (#38751)

Co-authored-by: Scott Beddall (from Dev Box) <scbedd@microsoft.com>

Commit:
null check (#38755)

JoshLove-msft committed:
Support infinite lock renewal (#38754)

* Support infinite lock renewal

* woops

* lower timeout

minhsuanlee committed:
New .NET SDK for RecoveryServices-DataReplication (#38340)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230915.2 (#38762)

ethanann-ms committed:
Updated control plane sdk with api 2023-06-30 (#38282)

RodgeFu committed:
update samplegen config for batchconfig (#38596)

htippanaboya committed:
NRP 2023-05-01 Release for .NET SDK (#38414)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Network (#38765)

saviesacov committed:
Marketplace - Upgrade to version 2023-01-01 (#38410)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Marketplace (#38766)

Ihab-Zhaika committed:
Azure Security Center: update pricings to 2023/01/01 (#38480)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.IotHub (#38770)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.SecurityCenter (#38771)

shivamverma-ms committed:
.NET SDK Release for JUNE ASR API VERSION (#38746)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230917.2 (#38778)

amis92 committed:
fix markdown link formatting in CONTRIBUTING.md (#38788)

yashjain4 committed:
Changes to generate updated SDK for new api version 2023-08-01 for Azure Resource Mover (#38660)

jsquire committed:
[Event Hubs] Disable extensions test (#38790)

Ignoring a function extension test that has been failing in CI until
it can be corrected.

alrod committed:
Adding [EditorBrowsable(EditorBrowsableState.Never)] for storage scal… (#38686)

amnguye committed:
[Storage] [DataMovement] Convert TransferStatus from enum to class (#38374)

* WIP

* WIP

* WIP

* WIP

* WIP - tests updated

* Finished fixing tests with changes

* Cleanup

* ExportApi

* WIP

* Fixes to live tests

* PR Comments - part 1

* Export API

* PR Comments

jaschrep-msft committed:
DataMovement File Share initial types (#38713)

* provider

* StorageResource types added

* construction of resources

* compile fixes

* remove blob mentions

* exportapi

* remove share resources, only directories

azure-sdk committed:
Serilaize with depth (#38757)

Co-authored-by: Daniel Jurek <djurek@microsoft.com>

heaths committed:
Add policy statement about code coverage (#38722)

* Add policy statement about code coverage

Resolves #37416

* Resolve PR feedback

jsquire committed:
[Event Hubs] Remove experimental package (#38792)

The focus of these changes is to remove the "experimental" package.

brrusino committed:
updated chaos SDK and tests (#38542)

* updated chaos SDK and tests

* updated assets and fixed merge issue in md

* Update autorest.md

* updating assets

* updating assets

* Updated assets, added delay on experiment record

* modified createVmssId, updated assets

* updated deployment bicep

* updated changelog

* Updated changelog

senatol committed:
Update SDK version from beta to GA for cosmosdbforpostgresql (#38317)

RodgeFu committed:
include NoValueResponse in shared code (#38380)

ArthurMa1978 committed:
Fix the ci.mgmt.yml (#38809)

RodgeFu committed:
move back NoValueResponseOfT to shared code which is needed for GetIfExists() method. (#38816)

archerzz committed:
sdk(anormalydetector): update spec version to fix syntax issue (#38818)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230919.1 (#38817)

Co-authored-by: Arcturus Zhang <dapzhang@microsoft.com>

ArthurMa1978 committed:
Update the test record for ResourceMover (#38819)

HarveyLink committed:
Inherent Check for ArmResource and ArmCollection (#38235)

christothes committed:
Update readme and troubleshooting guide (#38825)

jaschrep-msft committed:
test utilities (#38828)

* test utilities

* fix usings

amnguye committed:
[Storage] Storage Client Options Support Audiences (#38438)

* Initial work for Blob Clients to support Audience

* Export API

* Moved BlobAudience to Models; Added Audience support in DataLake; Addressed PR commeents

* Added audience support for Shares and Queues; Recorded tests; Cannot record tests for files audience tests

* Rerecord failure expected tests

* Adding default scope after resource id is set; Recorded Share Files tests

* Actually removing /.default from tests and correct order of what default to use

* Trying to resolve CI issues with Audience.Equals(string)

* Attempt to fix CI issues

* Cleanup

vishweshbankwar committed:
Timeout for httpclient for VM metadata service and suppress statsbeat instrumentation (#38801)

* timeout for httpclient and suppress instrument

* Update AzureMonitorExporterEventSource.cs

update EventSource message

---------

Co-authored-by: Timothy Mothra <tilee@microsoft.com>

vishweshbankwar committed:
Fix offline storage during network errors (#38832)

* Fix offline storage during network errors

* fix net462

* changelog

alrod committed:
Fix multiple triggers of the same type adding to DI (#38756)

* Fix multiple triggers of the same type adding to DI

* Compare with created instance

* Enabling ScaleHostEndToEndTest again

* Adding a comment

alrod committed:
Fix naming in AddAzureStorageQueuesScaleForTrigger (#38843)

JoshLove-msft committed:
Support MaxConcurrentCallsPerSession (#38836)

seanmcc-msft committed:
Storage STG 91 (#38839)

ArthurMa1978 committed:
Fix the ValidateInheritance naming test failures (#38844)

HarveyLink committed:
[RecoveryServicesBackup] Fix LRO issue (#38784)

* Add LRO

* Update CHANGELOG.md

* Update comments

* Add regression tests

* Update assets.json

* update

---------

Co-authored-by: Chengming <v-cruan@microsoft.com>
Co-authored-by: Wei Hu <huwe@microsoft.com>

ArthurMa1978 committed:
Fix the IPAddress serialization issue for Peering (#38824)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230920.1 (#38852)

* Update Generator Version 3.0.0-beta.20230920.1

* Update SDK codes

* Update SDK codes

* Update SDK codes

richardcho-msft committed:
Richardcho/cancel add participant (#38794)

* update swagger

* update swagger

* apply swagger updates

* update api

* Revert "apply swagger updates"

This reverts commit 99ebc379bacb3037ea52c4ad122cf9f91b82692e.

* auto generate

* add method

* add internal events

* update api

* update api

* update swagger

* auto generate

* add events

* add event processor changes

* fix build

* update api

* update code

* update typo

* fix typo

* add unit tests

* remove dupe file

* update swagger

* fix typo

* fix typo

* fix test

* address PR comments

---------

Co-authored-by: root <root@DESKTOP-6GBNLER>

Commit:
[AzureMonitorDistro] disable test (#38840)

* fix test

* investigating test...

* investigating tests

* disable test

Commit:
[AzureMonitorDistro] update OTel ResourceDetectors (#38833)

* update ResourceDetectors

* changelog

Commit:
[AzureMonitorExporter] prepare new release. Exporter 1.0.0. Distro beta7 (#38838)

* bump version

* update Readme

* fix date

* fix changelog header

* prepare Distro

jalauzon-msft committed:
[Storage][DataMovement] Add resource specific checkpoint data classes (#38842)

azure-sdk committed:
Increment version for monitor releases (#38864)

* Increment package version after release of Azure.Monitor.OpenTelemetry.Exporter

* Increment package version after release of Azure.Monitor.OpenTelemetry.AspNetCore

azure-sdk committed:
Increment package version after release of Azure.Identity (#38667)

Shashank1306s committed:
.NET SDK release changes for dataprotection (#38408)

* generating sdk for 2023-05-01

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* Addressing breaking change comment

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* Changes

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* Handling breaking changes

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* generating sdk for 2023-05-01

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* Addressing breaking change comment

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* Changes

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* Handling breaking changes

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* Updated DataProtectionBackupVaultProperties

* resolved breaking changes in ArmDataProtectionBackupModelFactory

* Add changelog

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* fixes

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* Renaming parameters

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* Update changelog

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* Added expected release date

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

* Renamed NamespacedNameResource

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>

---------

Signed-off-by: SHASHANK SINGH <shashasingh@microsoft.com>
Co-authored-by: SHASHANK SINGH <shashasingh@microsoft.com>
Co-authored-by: Himanshu Agarwal <hiaga@microsoft.com>

christothes committed:
Update MSAL dependencies to latest (#38866)

trrwilson committed:
Azure OpenAI: audio transcription and translation (#38460)

* squashed commit: whisper transcription/translation support

* Update to latest TypeSpec

* rebase codegen to mvp tsp PR

* Merge, snap, suppression cleanup

* PR feedback: remove errant <auto-generated/> tags in /custom

* Test refresh, incl. temporary BYOD rollback

* test fix for recordings and omitting multipart audio bodies

* PR feedback: idiomatic response formats

* PR feedback: fully distinguish translation and transcription types

* PR feedback: keep well-(enough)-known names for Srt,Vtt

* test: revert accidentally included local swap to live test mode

* full test update, including RAI adjustments

* merged .tsp snap (regen pending)

* code regen after merge and tsp snap

* CHANGELOG update

* Incorporate fabulous PR feedback. Thank you, Jose!

trrwilson committed:
CHANGELOG snap for beta.8 release (#38897)

metjuperry committed:
Added support for serialization of DataMemberAttribute in QueryAsync (#35954)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230920.3 (#38877)

Co-authored-by: Rodge Fu <wenjiefu@microsoft.com>

azure-sdk committed:
Autorest Regen Preview to 3.0.0-alpha.20230918.5 by Pan Shao from refs/pull/3671/head (#38813)

* Update Generator Version 3.0.0-alpha.20230918.5

* Update SDK codes

* update

* update

* update

---------

Co-authored-by: pshao25 <97225342+pshao25@users.noreply.github.com>

LeiWang3 committed:
Add tests for Servers API version 2023-02-01with new property IsIPv6Enabled (#38802)

* Add tests for Server IPv6 enabled

* Add generated .net swagger

* Add generated .net swagger

* Mitigate breaking changes

* Mitigate breaking changes

* Add seeet.json

* fix recording

* regen

---------

Co-authored-by: Wei Hu <huwe@microsoft.com>

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230921.2 (#38910)

* Update Generator Version 3.0.0-beta.20230921.2

* Update SDK codes

* update test of agrifood

---------

Co-authored-by: FAREAST\chunyu <chunyu@microsoft.com>

chunyu3 committed:
add a function to convert primitive type to RequestContent (#38808)

* add a function to convert primitive type to RequestContent

* add test

* add more test

azure-sdk committed:
Increment package version after release of Azure.AI.OpenAI (#38898)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20230922.2 (#38914)

jaschrep-msft committed:
DataMovement `ShareDirectoryStorageResourceContainer` implementation (#38892)

* resource container listing implementation

PathScanner for share directories.
Test PathScanner.

* tests

* getstorageresourcereference

* fix net462 issue

wiboris committed:
Update CODEOWNERS (#38903)

Updating Batch owners

christothes committed:
Update Tables changelog (#38899)

ShivangiReja committed:
[App Config] Regenerate code using API version "2023-10-01" (#38904)

* Update generated code

ShivangiReja committed:
[App Config] Rename snapshot name parameter to snapshotName (#38921)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 7002 (#38926)

* Add LockDeletionForDays parameter to set PodDisruptionBudget and cleanup job

* Use matrix for parallel tests. PDB improvements+docs.

* Fix kubectl namespace context preservation on login

---------

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

christothes committed:
Managed Identity retries on 410 status (#37012)

christothes committed:
Kubernetes workload identity test (#38708)

scottaddie committed:
Update XML docs for public Monitor Query APIs (#38902)

* Update XML docs for public Monitor Query APIs

* Update parameter descriptions

* Fix typo

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230922.4 (#38922)

* Update Generator Version 3.0.0-beta.20230922.4

* Update SDK codes

* Update SDK codes

* Update SDK codes

* Update SDK codes

* Update SDK codes

* Update SDK codes

* Update SDK codes

olegoid committed:
Devcenter dataplane SDK v2023-04-01, integration tests, and samples (#36245)

* updating generated code

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Sql (#38933)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.CosmosDBForPostgreSql (#38931)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ResourceMover (#38937)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 6919 (#38942)

* Add autorest-preview pipeline

* Add emitternpminstall

* Always install

* Use shorter leg name

* Add short circuiting to emitter install

* Use language matrix function

* Don't look for subfolders in the matrix package folders

* Revert unnecessary eng/common changes

* Rewrite GetPullRequestUrl to Get-BuildSourceDescription

* Remove alias from Invoke-LoggedCommand

* Use invoke-expression instead of shelling out

* Add better job splitting

* Replace Folder with Directory

---------

Co-authored-by: Patrick Hallisey <pahallis@microsoft.com>

amnguye committed:
Prepare for WebJobs Storage Extension Release (#38943)

kinelski committed:
[AppConfiguration] Fix: GetConfigurationSettings does not set the ContentType property (#38917)

cochi2 committed:
Adding Hold/Unhold capabilities to ACS. (#38858)

* Adding hold/unhold

* Updating swagger

* Fixing last merge

* Adding tests and missing apis

* Exporting new APIs

* Updating swagger

* Adding missing files

* Updating docstring

* Export API

* Updating swagger location

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230924.2 (#38934)

* Update Generator Version 3.0.0-beta.20230924.2

* Update SDK codes

* remove unused Etag customization

---------

Co-authored-by: FAREAST\chunyu <chunyu@microsoft.com>

HarveyLink committed:
[Network] Fix partial resource (#38110)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.DataProtectionBackup (#38953)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20230925.2 (#38955)

azure-sdk committed:
Increment version for storage releases (#38946)

* Increment package version after release of Microsoft.Azure.WebJobs.Extensions.Storage.Blobs

* Increment package version after release of Microsoft.Azure.WebJobs.Extensions.Storage.Queues

* Increment package version after release of Microsoft.Azure.WebJobs.Extensions.Storage

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Network (#38957)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.RecoveryServicesSiteRecovery (#38958)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Chaos (#38956)

frantran committed:
Fleet .NET SDK generation (#38873)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.DatabaseFleetManager (#38960)

jaschrep-msft committed:
DataMovement Test Infra: In-Memory StorageResources (#38941)

* memoryresources initial implementation

* revised implementaoin

* testing memoryitem

* test container

Jingshu918 committed:
[DataFactory]Upgrade SDK Version To 1.0.0-beta.4 (#38930)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.DataFactory (#38967)

dvbb committed:
[Network] Fix HopLink format error issue when resourceId return "" (#38876)

kinelski committed:
[FormRecognizer] Cleaning up misc operations live tests (#38807)

benbp committed:
Update messaging stress to use addons parallel job configuration (#38949)

rajkumar-rangaraj committed:
Vendor OpenTelemetry.NET Instrumentation Libraries into Azure.Monitor.OpenTelemetry.AspNetCore (#38848)

* Vendoring

* Update copyrights

* Update Notice file

rajkumar-rangaraj committed:
Remove the default OpenTelemetry instrumentation options customization from the README. (#38970)

rajkumar-rangaraj committed:
Remove OpenTelemetry ASP.NET Core and HTTP metrics. (#38972)

jaschrep-msft committed:
DataMovment `StorageResourceContainer` internal hooks to protected APIs (#38968)

* internals access class and hookup to file

* applied to blobs

* docsting

amnguye committed:
Implemention ShareFileStorageResource (#38938)

* WIP

* Implemented base file storage resource; Testst

* Export API

* Attempt to fix double dependency files in tests

* Addressing PR comments; Added StorageResourceItemInternal

* Rename internal methods

* PR Comments for tests

* CHange out verify parameters with  the mock verify method

hallipr committed:
Support typespec to sdk preview pipeline (#38576)

* Add changes to support typespec preview pipeline

* Revert whitespace changes

* Improve error logging in Update-dotnet-GeneratedSdks

scottaddie committed:
Update Monitor labels (#38974)

azure-sdk committed:
Change prod maintenance day. Fix stress watcher deploy (#38948)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

rohantagaru committed:
Azure Deployment Manager is being decommissioned. Remove its NET SDK through this PR (#38614)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20230926.2 (#38979)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230927.1 (#38987)

m-redding committed:
fix (#38995)

jsquire committed:
[AppConfig] Allow unknown condition members (#38971)

* [AppConfig] Allow unknown condition members

The focus of these changes is to allow for unknown members to be
present in the `conditions` property of a feature flag configuration
setting.  Previously, these were ignored as the setting assumed that
`conditions` should be treated as a well-known member.  This change moves
the well-known member to `conditions/client_filters` and preserves unknown
children of `conditions`.

seanmcc-msft committed:
Enabled STG 91 live tests (#38990)

m-redding committed:
[Core] Adding attributes to support Native AOT compiling (#37734)

* part 1

* additional changes

* additional attributes

* additional attributes

* adding diagnosticscope attributes

* updating messages

* more fixes

* additional updates

* updates

* PR feedback

* API update

* removing unnecessary warnings

* reverting attribute on HttpPipelineSynchronousPolicy

* update api

* updates

* update

* API export

* WIP fix 1

* WIP

* attribute adjustments

* adding compatibility test

* WIP adding test

* WIP API

* WIP

* WIP

* updates

* add test script and remove #if statements

* fixing build

* Update Program.cs

* adding some compile statements

* Update AzureCoreEventSource.cs

* build fix attempt

* fix built attempt 2

* Update MutableJsonChange.cs

* Update MutableJsonChange.cs

* Update MutableJsonDocument.cs

* Update HttpPipelineSynchronousPolicy.cs

* adding comments

* PR fb 1

* test

* test2

* test3

* attempt

* additional build fixes

* Remove comment

* PR feedback + fixing impacts of feedback changes

* fixes

* Update GeoRedundantFallbackPolicy.cs

* Update GeoRedundantFallbackPolicy.cs

* Update GeoRedundantFallbackPolicy.cs

* fix

* add tests

* pipeline fixes

* update comments

* feedback

* adjustments

* adjustment

* pipeline fix

* PR feedback 1

* Apply suggestions from code review

Co-authored-by: Vitek Karas <10670590+vitek-karas@users.noreply.github.com>

* PR feedback updates 2

* Update sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

Co-authored-by: Vitek Karas <10670590+vitek-karas@users.noreply.github.com>

* need one other suppression

* remove 2 annotations + update suppression message

* removing all attributes from RequestContentHelper

* tweak justification

* removing test project from Azure.Core, will be adding feedback to general CI pipeline

* WIP

* feedback

* Update sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

Co-authored-by: Anne Thompson <annelo@microsoft.com>

* fix attribute and add comments

* Addressing feedback

* adding net 5 preprocessor directives

* Update sdk/core/Azure.Core/src/DynamicData/MutableJsonElement.cs

Co-authored-by: Vitek Karas <10670590+vitek-karas@users.noreply.github.com>

* remove unused const

* pipeline fix

---------

Co-authored-by: Vitek Karas <10670590+vitek-karas@users.noreply.github.com>
Co-authored-by: Anne Thompson <annelo@microsoft.com>

rangothedog committed:
[Text Translation] Options Parameter for Translate and Transliterate (#38760)

* Options Parameter for Translate and Transliterate

* Remove empty options constructor, add new samples and recorded tests, update documentation.

* Update CHANGELOG.md and assets.json, fixed bug in TranslationLiveTests.cs

* Deleting unnecessary SessionRecords test files.

---------

Co-authored-by: Rango Meadows <v-rameadows@microsoft.com>

pshao25 committed:
Add rules for convenience method and protocol method (#38603)

* Add rules for convenience method and protocol method

* update

azure-sdk committed:
Increment version for managednetworkfabric releases (#39010)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230927.3 (#39013)

jaschrep-msft committed:
ShareDirectoryClient extensions (#38993)

* ShareDirectoryClient extensions

* removed mention of blobs

* exportapi

* fix tests

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20230928.1 (#39017)

vivekmore-msft committed:
Added transcription APIs and events (#38988)

* Autorest generated code

* Adding SDK methods for start, stop and update transcription

* Add transcription options for the create & answer call

* Adding events

* Adding event parser test cases

* Adding tests

* Refactoring code to make options param optional

* Adding models and parser for the transcription schema

* Fixing enum conversion logic for transcription schema

* Cleaning up code

* Add transcription options in create/answer/group call and tests

* Updted TranscriptionUpdate to take TranscriptionUpdate and TranscriptionStatusDetails as enums

* Fixed API typo

* Adding SDK methods for start, stop and update transcription

* Add transcription options for the create & answer call

* Adding models and parser for the transcription schema

* Adding events

* Adding event parser test cases

* Adding tests

* Refactoring code to make options param optional

* Fixing enum conversion logic for transcription schema

* Cleaning up code

* Add transcription options in create/answer/group call and tests

* Updted TranscriptionUpdate to take TranscriptionUpdate and TranscriptionStatusDetails as enums

* Fixed API typo

* Updated contract after rebase

* Removed duplicate tests

* Update in transcription ws schema

* Fixed TranscriptionOptions review comment

* Fixing TranscriptionOptions arguments in create call, group call & answer call

* Fixing build issues

---------

Co-authored-by: abhishesingh-msft <abhishesingh@microsoft.com>

seanmcc-msft committed:
Modified storage ARM template to point at built-in Files RBAC role (#39018)

JoshLove-msft committed:
Add support for settlement from the isolated worker extension (#38865)

* Add support for settlement from the isolated worker extension

* Update comment

* roll back version of Grpc.Tools

* roll back further

* PR fb

* Revert eventArgs fields to private

* remove in finally block

* Fix test and move versions to package.data.props

* Update sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Add batch test cases

* Fix tests

* Fix

---------

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

ArthurMa1978 committed:
Update the migration guide. (#38961)

glecaros committed:
[ARM] Updating Azure.ResourceManager.MachineLearning to generate for … (#38814)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.MachineLearning (#39035)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.RecoveryServicesDataReplication (#39034)

JimSuplizio committed:
add traling slash to directory wildcard entries (#39038)

JimSuplizio committed:
Update CODEOWNERS (#39039)

Just testing to see if this path isn't recognizing owners because the ones there were invalid

weshaggard committed:
Revert "Update CODEOWNERS (#39039)" (#39040)

This reverts commit ec8e5bb1cfb86d8824de5ac3c3973dc1e920fd1c.

benbp committed:
Remove old check enforcer comment template (#39042)

jsquire committed:
[JimBot] Remove CXP Attention (#39036)

Removing config for the dead CXP Attention rule

christothes committed:
Multi-tenant tests added as recorded tests (#38992)

christothes committed:
TokenExchangeManagedIdentitySource with async IO (#38939)

heaths committed:
Add AOT target for Key Vault SDK (#38721)

* Add AOT target for Key Vault SDK

Resolves #38500

* Rebase on main, resolve PR feedback

rajkumar-rangaraj committed:
Detect Instrumentation Package and Backoff  (#39001)

* AddVendorInstrumentationIfPackageNotReferenced

* PR feedback.

* event souce fix.

amnguye committed:
Replace LocalFilesStorageResourceProvider.FromPath with FromFile and Directory (#39053)

* Replace LocalFilesStorageResourceProvider.FromPath with FromFile and FromDirectory

* Export API

* Update snippets

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 7040 (#39052)

* Remove codeowners resolution logic from docs scripts

* Remove parameters that aren't needed

* Remove auth information from update-docsms-metadata.yml

---------

Co-authored-by: Daniel Jurek <djurek@microsoft.com>

heaths committed:
Add support for Key Vault 7.5-preview.1 (#38377)

* Add support for Key Vault 7.5-preview.1

Also treats KeyRotationPolicyAction as case-insensitive.

* Update public APIs

* Use latest commit

Resolves #38729

yaotongms committed:
fix compiler error

yaotongms committed:
fix compiler error

yaotongms committed:
fix compiler error

yaotongms committed:
update samples

yaotongms committed:
upload test recordings

yaotongms commented:
Test recordings uploaded to: https://github.com/Azure/azure-sdk-assets/tree/net/hybridcompute/Azure.ResourceManager.HybridCompute_8d2d4b8453

yaotongms committed:
fix invalid token

yaotongms committed:
update tests

yaotongms committed:
fix tests

yaotongms committed:
fix comments

yaotongms committed:
Merge branch 'hybridcompute' of https://github.com/yaotongms/azure-sdk-for-net into hybridcompute
